Documentation/en-us/TestUsingTestDoubles.md

```diff
@@ -0,0 +1,131 @@
+# Testing with Mock
+
+## Test doubles
+
+Following problem comes up often when writing tests. Here, `B` Class uses(depends on) `A Class`.
+
+![](https://github.com/Quick/Assets/blob/master/Screenshots/TestUsingMock_BusesA.png)
+
+Here's one example: BTests tests B, which uses A. Now bugs in A cause BTests to fail, even though B is fine! 
+All the failures are confusing--it's hard to answer the question "what's broken?"
+
+To avoid this problem, we can use a stand-in for A in Btests. In this case, we call stand-in for A `A' class`.
+
+![](https://github.com/Quick/Assets/blob/master/Screenshots/TestUsingMock_BusesAmock.png)
+
+`A' class` has same methods--however implementation might be different, and properties as `A Class`. So we can replace `A Class` with `A' Class` in BTests.
```

Comment:
This sentence is a little confusing I think. Perhaps better as "`A' class` has the same methods and properties as `A class` (although its implementations of those methods and properties might be different). So we can..."